### PR TITLE
Fixed Bugs #31 #32 #33

### PR DESCRIPTION
**Pull request by ravipwaghmare**
Fixed Following Bugs:
https://github.com/metal3-io/hardware-classification-controller/issues/31
https://github.com/metal3-io/hardware-classification-controller/issues/32
https://github.com/metal3-io/hardware-classification-controller/issues/33

@maelk @digambar15  
Kindly Review.

### CODE REVIEW

**Review by maelk — COMMENTED** [review]
looks good to me

File: controllers/hardwareclassification_controller.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func (hcReconciler *HardwareClassificationReconciler) Reconcile(req ctrl.Request
 	ctx := context.Background()
 
 	// Initialize the logger with namespace
-	hcReconciler.Log = hcReconciler.Log.WithName(HWControllerName).WithValues("metal3-hardwareclassification", req.NamespacedName)
+	hccLog := hcReconciler.Log.WithName(HWControllerName).WithValues("metal3-hardwareclassification", req.NamespacedName)
```
Comment (digambar15) [inline controllers/hardwareclassification_controller.go]: change it to hwclog

**Review by digambar15 — CHANGES_REQUESTED** [review]

File: controllers/hardwareclassification_controller.go (hunk 1 above)
Comment (ravipwaghmare) [reply controllers/hardwareclassification_controller.go]: Done

**Comment by digambar15** [comment]
Looks good.

**Review by digambar15 — APPROVED** [review]

**Pull request merged**